packages/proxy/lib/http/response-middleware.ts

```diff
@@ -431,17 +430,17 @@ const CopyCookiesFromIncomingRes: ResponseMiddleware = async function () {
   //   clear that it's one we're handling ourselves.
   // - We also set the cookies through automation so they are available in the
   //   browser via document.cookie and via Cypress cookie APIs
-  //   (e.g. cy.getCookie). This is only done for cross-origin responses, since
-  //   non-cross-origin responses will be successfully set in the browser
+  //   (e.g. cy.getCookie). This is only done for third-party cookie context responses, since
+  //   first party cookie responses will be successfully set in the browser
   //   automatically.
   // - In the request middleware, we retrieve the cookies for a given URL
   //   and attach them to the request, like the browser normally would.
   //   tough-cookie handles retrieving the correct cookies based on domain,
   //   path, etc. It also removes cookies from the cookie jar if they've expired.
-  const needsCrossOriginHandling = checkIfNeedsCrossOriginHandling(this)
+  const needsThirdPartyHandling = checkIfCookieNeedsThirdPartyHandling(this)
 
   const appendCookie = (cookie: string) => {
-    const headerName = needsCrossOriginHandling ? 'X-Set-Cookie' : 'Set-Cookie'
+    const headerName = needsThirdPartyHandling ? 'X-Set-Cookie' : 'Set-Cookie'
```

Comment:
nit: Isn't `needsCrossOriginHandling` still representative here? I assume it could be third part, cross origin and/or both?

Reply:
I figured it would be `thirdParty` since the browser should be able to set first party cookies even on _some_ cross origin requests. That being said I am not thrilled about the name I have here 😅 .

Reply:
resolving this as it is no longer applicable